src/wallet/wallet.h

```diff
@@ -700,7 +700,16 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
 
     bool fFileBacked;
 
-    std::set<int64_t> setKeyPool;
+    // class for holding a keypool-key-entry in memory
+    class KeypoolCache {
+    public:
+        int64_t nIndex;
+        CKeyID keyId;
```

Comment:
Yes. This is true, though, the keyId is relatively important for future performance improvements. We could add it later,.. but I guess it doesn't hurt to already have it there in this PR.